### DISCUSSION TITLE
Populate multiple text boxes from a dropdown

### DISCUSSION THREAD

**Discussion by bigfoot53** [Budibase/budibase#9883] [Help] [answered]
Hello Everyone, 
I'm looking for a way to get a drop-down to populate up to 6 text boxes . What i have so far is a db that holds the selections for the drop-down  it has 2 data points one for tool and the other for uses  so teach tool can have up to 6 uses. i have a second db that is for the general data entry  and contains at uses text field . what i would like to do is create a form where i can  have the use select the tool from the drop-down "Connected to the Tool DB" and it will populate the 6 uses fields  and allow the use to continue filling out the form and once saved the data will all be saved in the general db   Is this possible  ? 

Thanks
Bruce

**Comment by melohagan** [comment]
Hey @bigfoot53 

I didn't quite follow this part:

> What i have so far is a db that holds the selections for the drop-down it has 2 data points one for tool and the other for uses so teach tool can have up to 6 uses. i have a second db that is for the general data entry and contains at uses text field . what i would like to do is create a form where i can have the use select the tool from the drop-down "Connected to the Tool DB" and it will populate the 6 uses fields and allow the use to continue filling out the form and once saved the data will all be saved in the general db Is this possible ?

Perhaps some screenshots and a step by step description of the problem would help!

**Comment by bigfoot53** [comment]
Hey @melohagan 
Thanks for responding, 
Im currently traveling and dont have access to the db  to get screen shots  but ill try to describe the best that i can,

I’m creating  an evaluation form that will allow the user to rate interactions and or  skills for different tools. Each tool has a different set “up to 6 different skills or interactions” that are based on the tool used. Here is what I would like to achieve.   I would like to have a picker “dropdown”  that allows the user to select a tool  and that selection would populate the skills int the form  where the user can then finish filling out the form and submit. The picker would pull the information from a db called tools it contains 2 columns  skill and tool. The main form stores its data in a different db  which holds the selected data form the picker as well as the remaining data provided by the user.
Since I don’t have access to my db I created the attached pic as a reference  
 For reference, the light blue  skills are populated by the picker “tool Select” which gets its data from the tools db.  The grey scores are populated by the user  in a range of 1 to 5 .  the peachish fields are just other user data  they have  enter. Clicking the submit  either progresses to the next step of the form or enters the current data into the main database and clears the form, 
I hope this helps 😊 
 
 


![DB2](https://user-images.githubusercontent.com/61506151/223302909-ffc58b8a-679a-4103-a01b-0c15e47aa9f6.png)

**Reply by melohagan** [reply]
Thanks for the extra info @bigfoot53 

So my understanding then is that you just have one table here called *Tools* and it has two text columns: *Skill* and *Tool Name*

Do you have a *Skills* table as well? What does the relationship looks like, is it One Tool -> Many Skills?

If you only have the one *Tools* table, then how would you have multiple skills?

**Reply by bigfoot53** [reply]
2 tables, tools table holds the data points for the main form  "shift, user, score, etc   the Skills table holds the info the picker uses  2 columns  one for the tool name and one for the skill associate with the tool ,  yes the relationship is one tool to many skills  :)

**Reply by ConorWebb96** [reply]
Hey @bigfoot53, would we be able to trouble you for an app export so we can take a more indepth look.

**Comment by bigfoot53** [comment]
Sure, Ill work on that when i get back this weekend

**Comment by bigfoot53** [comment]
[My app-export-1678586279094.tar.gz](https://github.com/Budibase/budibase/files/10949852/My.app-export-1678586279094.tar.gz)
Evening everyone, here is the export . i did notice that the form isnt displaying now its just a blank page  :(   probably something i did but not sure

**Comment by ConorWebb96** [comment]
Hey @bigfoot53, 

There was a lot of issues with what you sent across. The first one being that you dragged the form and related elements outside of the data provider. Thats why nothing was showing.

I used your export as a framework to work off and started from scratch taking elements I thought were necessary.

I've created two new tables to separate the skills and tools. (I've added relationships between these two tables) I tried to stick to your table structure for the shifts although I didn't really understand some of the fields within i've tried to work with it as best I can.

**Sample Images of this working**
<img width="1399" alt="Screenshot 2023-03-13 at 15 06 48" src="https://user-images.githubusercontent.com/126772285/224742888-1b102bb5-669c-4965-9640-1b3a10926fd2.png">

<img width="1393" alt="Screenshot 2023-03-13 at 15 06 54" src="https://user-images.githubusercontent.com/126772285/224742957-0f4ad2e6-02e8-4f9d-a8fb-6b0d92d5aab1.png">

The below export is a working version of what you requested. (from what I understand)
[multi tables-export-1678719581505.tar.gz](https://github.com/Budibase/budibase/files/10958714/multi.tables-export-1678719581505.tar.gz)

Hopefully this helps!

**Comment by bigfoot53** [comment]
@ConorWebb96 
Thanks for the updated version  i have added a few more tools and associated skills. Ive noticed that when i select the tool the skills fill in but they are not correct  and everything after the 1st answer is a duplicate.  How can i correct this ?
![image](https://user-images.githubusercontent.com/61506151/225188621-925e5954-ba9e-4140-affd-77db8f7da4fe.png)
![image](https://user-images.githubusercontent.com/61506151/225188992-e9843112-52b2-431d-a1d5-2bccf95111fd.png)


Thanks
Bruce

**Comment by ConorWebb96** [comment, marked as answer]
Hey @bigfoot53,

I've fixed the issues, was a mistake on my end when testing I didn't have enough data and saw it changing thinking it was working!
The below should now work for you.

[multi tables-export-1678878755841.tar.gz](https://github.com/Budibase/budibase/files/10978967/multi.tables-export-1678878755841.tar.gz)

Hopefully this helps!

**Comment by bigfoot53** [comment]
Awesome, Thanks very much . I will be adding a few more columns to the db and i see that when you add a column it puts it at the end.  is there a way to re-arrange the columns  ?

**Reply by ConorWebb96** [reply]
Hey @bigfoot,

There isn't a way currently to rearrange the table columns in the data section. You can use table component in design view to Configure columns and change them around.

Hopefully this helps!
